runtime/lua/vim/lsp/diagnostic.lua

```diff
@@ -110,14 +110,24 @@ local function diagnostic_lsp_to_vim(diagnostics, bufnr, client_id)
   return vim.tbl_map(function(diagnostic)
     local start = diagnostic.range.start
     local _end = diagnostic.range['end']
+    local message, message_kind --- @type string, lsp.MarkupKind
+    if type(diagnostic.message) == 'string' then
```

Comment:
is type(diagnostic.message) == 'string' check is there to handle data from language servers that's not following 3.18 or is this no longer applicable?

Reply:
@s1n7ax Version 3.18 of the spec still says that diagnostic messages can be strings:
```
/**
  * The diagnostic's message.
  *
  * @since 3.18.0 - support for MarkupContent. This is guarded by the client
  * capability `textDocument.diagnostic.markupMessageSupport`.
  */
  message: string | MarkupContent
```